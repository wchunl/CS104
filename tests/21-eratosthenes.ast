ROOT "" (0.0.0)
|   TOK_FUNCTION "" (6.22.1) function
|   |   TOK_TYPE_ID "" (6.22.1) typeid
|   |   |   TOK_VOID "void" (6.22.1) void
|   |   |   TOK_IDENT "fail" (6.22.6) variable
|   |   TOK_PARAM "(" (6.22.11) param
|   |   |   TOK_TYPE_ID "" (6.22.12) typeid
|   |   |   |   TOK_STRING "string" (6.22.12) string
|   |   |   |   TOK_IDENT "expr" (6.22.19) string variable (6.22.12)
|   |   |   TOK_TYPE_ID "" (6.22.25) typeid
|   |   |   |   TOK_STRING "string" (6.22.25) string
|   |   |   |   TOK_IDENT "file" (6.22.32) string variable (6.22.25)
|   |   |   TOK_TYPE_ID "" (6.22.38) typeid
|   |   |   |   TOK_INT "int" (6.22.38) int
|   |   |   |   TOK_IDENT "line" (6.22.42) int variable (6.22.38)
|   TOK_FUNCTION "" (6.24.1) function
|   |   TOK_TYPE_ID "" (6.24.1) typeid
|   |   |   TOK_VOID "void" (6.24.1) void
|   |   |   TOK_IDENT "putchr" (6.24.6) variable
|   |   TOK_PARAM "(" (6.24.13) param
|   |   |   TOK_TYPE_ID "" (6.24.14) typeid
|   |   |   |   TOK_INT "int" (6.24.14) int
|   |   |   |   TOK_IDENT "chr" (6.24.18) int variable (6.24.14)
|   TOK_FUNCTION "" (6.25.1) function
|   |   TOK_TYPE_ID "" (6.25.1) typeid
|   |   |   TOK_VOID "void" (6.25.1) void
|   |   |   TOK_IDENT "putint" (6.25.6) variable
|   |   TOK_PARAM "(" (6.25.13) param
|   |   |   TOK_TYPE_ID "" (6.25.14) typeid
|   |   |   |   TOK_INT "int" (6.25.14) int
|   |   |   |   TOK_IDENT "num" (6.25.18) int variable (6.25.14)
|   TOK_FUNCTION "" (6.26.1) function
|   |   TOK_TYPE_ID "" (6.26.1) typeid
|   |   |   TOK_VOID "void" (6.26.1) void
|   |   |   TOK_IDENT "putstr" (6.26.6) variable
|   |   TOK_PARAM "(" (6.26.13) param
|   |   |   TOK_TYPE_ID "" (6.26.14) typeid
|   |   |   |   TOK_STRING "string" (6.26.14) string
|   |   |   |   TOK_IDENT "str" (6.26.21) string variable (6.26.14)
|   TOK_FUNCTION "" (6.28.1) function
|   |   TOK_TYPE_ID "" (6.28.1) typeid
|   |   |   TOK_INT "int" (6.28.1) int
|   |   |   TOK_IDENT "getchr" (6.28.5) variable
|   |   TOK_PARAM "(" (6.28.11) param
|   TOK_FUNCTION "" (6.29.1) function
|   |   TOK_TYPE_ID "" (6.29.1) typeid
|   |   |   TOK_STRING "string" (6.29.1) string
|   |   |   TOK_IDENT "getstr" (6.29.8) variable
|   |   TOK_PARAM "(" (6.29.14) param
|   TOK_FUNCTION "" (6.30.1) function
|   |   TOK_TYPE_ID "" (6.30.1) typeid
|   |   |   TOK_STRING "string" (6.30.1) string
|   |   |   TOK_IDENT "getln" (6.30.8) variable
|   |   TOK_PARAM "(" (6.30.13) param
|   TOK_FUNCTION "" (7.8.1) function
|   |   TOK_TYPE_ID "" (7.8.1) typeid
|   |   |   TOK_INT "int" (7.8.1) int
|   |   |   TOK_IDENT "main" (7.8.5) variable
|   |   TOK_PARAM "(" (7.8.9) param
|   |   TOK_BLOCK "{" (7.8.12)
|   |   |   TOK_VARDECL "=" (7.9.14) variable lval
|   |   |   |   TOK_INT "int" (7.9.4) int
|   |   |   |   TOK_IDENT "prime" (7.9.8) int variable (7.9.14)
|   |   |   |   TOK_INTCON "2" (7.9.16) int const
|   |   |   TOK_VARDECL "=" (7.10.14) variable lval
|   |   |   |   TOK_INT "int" (7.10.4) int
|   |   |   |   TOK_IDENT "index" (7.10.8) int variable (7.10.14)
|   |   |   |   TOK_INTCON "2" (7.10.16) int const
|   |   |   TOK_VARDECL "=" (7.11.21) variable lval
|   |   |   |   TOK_ARRAY "array" (7.11.4) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.11.10) typeid
|   |   |   |   TOK_IDENT "sieve" (7.11.15) int variable (7.11.21)
|   |   |   |   TOK_ALLOC "alloc" (7.11.23)
|   |   |   |   |   TOK_ARRAY "array" (7.11.29) array
|   |   |   |   |   TOK_TYPE_ID "int" (7.11.35) typeid
|   |   |   |   |   TOK_INTCON "100" (7.11.41) int const
|   |   |   TOK_WHILE "while" (7.13.4)
|   |   |   |   '<' "<" (7.13.17)
|   |   |   |   |   TOK_IDENT "index" (7.13.11) int variable (7.10.14)
|   |   |   |   |   TOK_INTCON "100" (7.13.19) int const
|   |   |   |   TOK_BLOCK "{" (7.13.24)
|   |   |   |   |   '=' "=" (7.14.20) int vreg
|   |   |   |   |   |   TOK_INDEX "[" (7.14.12) lval vaddr
|   |   |   |   |   |   |   TOK_IDENT "sieve" (7.14.7) int variable (7.11.21)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.14.13) int variable (7.10.14)
|   |   |   |   |   |   TOK_INTCON "1" (7.14.22) int const
|   |   |   |   |   '=' "=" (7.15.13) int vreg
|   |   |   |   |   |   TOK_IDENT "index" (7.15.7) int variable (7.10.14)
|   |   |   |   |   |   '+' "+" (7.15.21) int vreg
|   |   |   |   |   |   |   TOK_IDENT "index" (7.15.15) int variable (7.10.14)
|   |   |   |   |   |   |   TOK_INTCON "1" (7.15.23) int const
|   |   |   TOK_WHILE "while" (7.18.4)
|   |   |   |   '<' "<" (7.18.17)
|   |   |   |   |   TOK_IDENT "prime" (7.18.11) int variable (7.9.14)
|   |   |   |   |   TOK_INTCON "100" (7.18.19) int const
|   |   |   |   TOK_BLOCK "{" (7.18.24)
|   |   |   |   |   TOK_IF "if" (7.19.7)
|   |   |   |   |   |   TOK_INDEX "[" (7.19.16) lval vaddr
|   |   |   |   |   |   |   TOK_IDENT "sieve" (7.19.11) int variable (7.11.21)
|   |   |   |   |   |   |   TOK_IDENT "prime" (7.19.17) int variable (7.9.14)
|   |   |   |   |   |   TOK_BLOCK "{" (7.19.25)
|   |   |   |   |   |   |   '=' "=" (7.20.16) int vreg
|   |   |   |   |   |   |   |   TOK_IDENT "index" (7.20.10) int variable (7.10.14)
|   |   |   |   |   |   |   |   '*' "*" (7.20.24) int vreg
|   |   |   |   |   |   |   |   |   TOK_IDENT "prime" (7.20.18) int variable (7.9.14)
|   |   |   |   |   |   |   |   |   TOK_INTCON "2" (7.20.26) int const
|   |   |   |   |   |   |   TOK_WHILE "while" (7.21.10)
|   |   |   |   |   |   |   |   '<' "<" (7.21.23)
|   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.21.17) int variable (7.10.14)
|   |   |   |   |   |   |   |   |   TOK_INTCON "100" (7.21.25) int const
|   |   |   |   |   |   |   |   TOK_BLOCK "{" (7.21.30)
|   |   |   |   |   |   |   |   |   '=' "=" (7.22.26) int vreg
|   |   |   |   |   |   |   |   |   |   TOK_INDEX "[" (7.22.18) lval vaddr
|   |   |   |   |   |   |   |   |   |   |   TOK_IDENT "sieve" (7.22.13) int variable (7.11.21)
|   |   |   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.22.19) int variable (7.10.14)
|   |   |   |   |   |   |   |   |   |   TOK_INTCON "0" (7.22.28) int const
|   |   |   |   |   |   |   |   |   '=' "=" (7.23.19) int vreg
|   |   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.23.13) int variable (7.10.14)
|   |   |   |   |   |   |   |   |   |   '+' "+" (7.23.27) int vreg
|   |   |   |   |   |   |   |   |   |   |   TOK_IDENT "index" (7.23.21) int variable (7.10.14)
|   |   |   |   |   |   |   |   |   |   |   TOK_IDENT "prime" (7.23.29) int variable (7.9.14)
|   |   |   |   |   '=' "=" (7.26.13) int vreg
|   |   |   |   |   |   TOK_IDENT "prime" (7.26.7) int variable (7.9.14)
|   |   |   |   |   |   '+' "+" (7.26.21) int vreg
|   |   |   |   |   |   |   TOK_IDENT "prime" (7.26.15) int variable (7.9.14)
|   |   |   |   |   |   |   TOK_INTCON "1" (7.26.23) int const
|   |   |   '=' "=" (7.29.10) int vreg
|   |   |   |   TOK_IDENT "index" (7.29.4) int variable (7.10.14)
|   |   |   |   TOK_INTCON "2" (7.29.12) int const
|   |   |   TOK_WHILE "while" (7.30.4)
|   |   |   |   '<' "<" (7.30.17)
|   |   |   |   |   TOK_IDENT "index" (7.30.11) int variable (7.10.14)
|   |   |   |   |   TOK_INTCON "100" (7.30.19) int const
|   |   |   |   TOK_BLOCK "{" (7.30.24)
|   |   |   |   |   TOK_IF "if" (7.31.7)
|   |   |   |   |   |   TOK_INDEX "[" (7.31.16) lval vaddr
|   |   |   |   |   |   |   TOK_IDENT "sieve" (7.31.11) int variable (7.11.21)
|   |   |   |   |   |   |   TOK_IDENT "index" (7.31.17) int variable (7.10.14)
|   |   |   |   |   |   TOK_BLOCK "{" (7.31.25)
|   |   |   |   |   |   |   TOK_CALL "(" (7.32.17)
|   |   |   |   |   |   |   |   TOK_IDENT "putint" (7.32.10) variable
|   |   |   |   |   |   |   |   TOK_IDENT "index" (7.32.18) int variable (7.10.14)
|   |   |   |   |   |   |   TOK_CALL "(" (7.33.17)
|   |   |   |   |   |   |   |   TOK_IDENT "putchr" (7.33.10) variable
|   |   |   |   |   |   |   |   TOK_CHARCON "'\n'" (7.33.18) int const
|   |   |   |   |   '=' "=" (7.35.13) int vreg
|   |   |   |   |   |   TOK_IDENT "index" (7.35.7) int variable (7.10.14)
|   |   |   |   |   |   '+' "+" (7.35.21) int vreg
|   |   |   |   |   |   |   TOK_IDENT "index" (7.35.15) int variable (7.10.14)
|   |   |   |   |   |   |   TOK_INTCON "1" (7.35.23) int const
|   |   |   TOK_RETURN "return" (7.38.4)
|   |   |   |   TOK_INTCON "0" (7.38.11) int const
