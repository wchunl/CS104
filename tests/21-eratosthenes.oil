fail:     .function
          .param ptr expr
          .param ptr file
          .param int line
          .end
putchr:   .function
          .param int chr
          .end
putint:   .function
          .param int num
          .end
putstr:   .function
          .param ptr str
          .end
getchr:   .function int
          .end
getstr:   .function
          .end
getln:    .function
          .end
main:     .function int
          .local int prime
          .local int index
          .local ptr sieve
          prime = 2
          index = 2
          malloc(4 * sizeof int)
          sieve = $t0:i
.wh0:     $t1:i = index < 100
          goto .od0 if not $t1:i
.do0:     sieve[index * :p] = 1
          $t2:i = index + 1
          index = $t2:i
.od0:     
.wh1:     $t3:i = prime < 100
          goto .od1 if not $t3:i
.do1:     
.if0:     $t4:p = sieve[prime * :p]
          goto .el0 if not $t4:i
.th0:     $t5:i = prime * 2
          index = $t5:i
.wh2:     $t6:i = index < 100
          goto .od2 if not $t6:i
.do2:     sieve[index * :p] = 0
          $t7:i = index + prime
          index = $t7:i
.od2:     
.fi0:     $t8:i = prime + 1
          prime = $t8:i
.od1:     index = 2
.wh3:     $t9:i = index < 100
          goto .od3 if not $t9:i
.do3:     
.if1:     $t10:p = sieve[index * :p]
          goto .el1 if not $t10:i
.th1:     putint(index)
          putchr('\n')
.fi1:     $t11:i = index + 1
          index = $t11:i
.od3:     return 0
          return 
          .end
