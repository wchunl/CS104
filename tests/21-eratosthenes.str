string_set[   0]:    4706206694200543916 0x15b6c28->","
                   16579916199511856337 0x15b80b8->"getln"
string_set[   1]:    2297668033614959926 0x15ba118->"0"
                    6637313742931709005 0x15b9c38->"*"
string_set[   3]:   11868120863286193964 0x15b82d8->"main"
                    9611428142580219387 0x15b25a8->"fail"
string_set[   4]:      21606623943142546 0x15b8538->"{"
                    4017852018407011111 0x15b6d18->"file"
string_set[   6]:   13686721134187162920 0x15b9568->"}"
string_set[   8]:    1370730858159036685 0x15b96e8->"+"
                    9386170080689500517 0x15b6af8->"expr"
string_set[   9]:   10319632053171904911 0x15b8b28->"while"
                   14526079891305879776 0x15b8cd8->">"
string_set[  11]:   17581720658427701949 0x15b8d48->"sieve"
string_set[  12]:   15816506932648033945 0x15b2228->"("
string_set[  13]:    6340238197210063670 0x15b90b8->"100"
string_set[  14]:    9317282556992397945 0x15b9858->"if"
string_set[  17]:   18413295556186398326 0x15b7118->"putint"
string_set[  18]:    7211130878751977441 0x15b7248->"putchr"
                   12679825168387489523 0x15b7728->"num"
string_set[  21]:   13581035385482130682 0x15b9178->"["
string_set[  23]:   12051031859636400188 0x15b9408->"]"
                   11166701219533395393 0x15b7e78->"getstr"
                   12868445110721718657 0x15b6d38->")"
string_set[  28]:   14635384904416252381 0x15bb208->"return"
string_set[  29]:   12291392672316406109 0x15b6ec8->"line"
string_set[  31]:    6192799629375248351 0x15baf78->"'\n'"
                    6142509188972423790 0x15b2588->""
                   13348923631108115385 0x15b7508->"putstr"
string_set[  32]:   11597697714117577063 0x15b87a8->"="
string_set[  33]:    8290819294854170237 0x15b8cf8->"alloc"
                    1583576014053361350 0x15b78a8->"str"
string_set[  38]:   12597418120160561823 0x15b6b68->"int"
string_set[  39]:    7296690267450279874 0x15b7388->"chr"
string_set[  40]:   11911796782241914524 0x15b6688->"void"
string_set[  41]:    1894629132964850000 0x15b7058->"prime"
string_set[  42]:    4698864730501174006 0x15b89c8->"index"
                    5344662657899890615 0x15b6ee8->";"
string_set[  43]:    6814028903552918164 0x15b88e8->"array"
                    4551451650890805270 0x15b8848->"2"
string_set[  44]:   13497578857207247412 0x15b6a28->"string"
string_set[  45]:   10159970873491820195 0x15b9428->"1"
                    1396094175042499165 0x15b8be8->"<"
string_set[  46]:    9201800968376702500 0x15b7c58->"getchr"
load_factor = 0.936
bucket_count = 47
max_bucket_size = 3
