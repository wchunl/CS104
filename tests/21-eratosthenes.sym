fail (6.22.1) {0} void function
   expr (6.22.12) {1} string variable param lval 0
   file (6.22.25) {1} string variable param lval 1
   line (6.22.38) {1} int variable param lval 2

putchr (6.24.1) {1} void function
   chr (6.24.14) {1} int variable param lval 0

putint (6.25.1) {2} void function
   num (6.25.14) {1} int variable param lval 0

putstr (6.26.1) {3} void function
   str (6.26.14) {1} string variable param lval 0

getchr (6.28.1) {4} int function

getstr (6.29.1) {5} string function

getln (6.30.1) {6} string function

main (7.8.1) {7} int function
   prime (7.9.14) {1} int variable local lval 0
   index (7.10.14) {1} int variable local lval 1
   sieve (7.11.21) {1} array <int> variable local lval 2
