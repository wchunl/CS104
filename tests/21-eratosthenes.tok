#  1 "21-eratosthenes.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "21-eratosthenes.oc"
#  1 "oclib.h"
# 22 "oclib.h"
   6 22.00001 261 TOK_VOID      void
   6 22.00006 282 TOK_IDENT     fail
   6 22.00011  40 '('           (
   6 22.00012 263 TOK_STRING    string
   6 22.00019 282 TOK_IDENT     expr
   6 22.00023  44 ','           ,
   6 22.00025 263 TOK_STRING    string
   6 22.00032 282 TOK_IDENT     file
   6 22.00036  44 ','           ,
   6 22.00038 262 TOK_INT       int
   6 22.00042 282 TOK_IDENT     line
   6 22.00046  41 ')'           )
   6 22.00047  59 ';'           ;
   6 24.00001 261 TOK_VOID      void
   6 24.00006 282 TOK_IDENT     putchr
   6 24.00013  40 '('           (
   6 24.00014 262 TOK_INT       int
   6 24.00018 282 TOK_IDENT     chr
   6 24.00021  41 ')'           )
   6 24.00022  59 ';'           ;
   6 25.00001 261 TOK_VOID      void
   6 25.00006 282 TOK_IDENT     putint
   6 25.00013  40 '('           (
   6 25.00014 262 TOK_INT       int
   6 25.00018 282 TOK_IDENT     num
   6 25.00021  41 ')'           )
   6 25.00022  59 ';'           ;
   6 26.00001 261 TOK_VOID      void
   6 26.00006 282 TOK_IDENT     putstr
   6 26.00013  40 '('           (
   6 26.00014 263 TOK_STRING    string
   6 26.00021 282 TOK_IDENT     str
   6 26.00024  41 ')'           )
   6 26.00025  59 ';'           ;
   6 28.00001 262 TOK_INT       int
   6 28.00005 282 TOK_IDENT     getchr
   6 28.00011  40 '('           (
   6 28.00012  41 ')'           )
   6 28.00013  59 ';'           ;
   6 29.00001 263 TOK_STRING    string
   6 29.00008 282 TOK_IDENT     getstr
   6 29.00014  40 '('           (
   6 29.00015  41 ')'           )
   6 29.00016  59 ';'           ;
   6 30.00001 263 TOK_STRING    string
   6 30.00008 282 TOK_IDENT     getln
   6 30.00013  40 '('           (
   6 30.00014  41 ')'           )
   6 30.00015  59 ';'           ;
#  4 "21-eratosthenes.oc"
   7 8.00001 262 TOK_INT       int
   7 8.00005 282 TOK_IDENT     main
   7 8.00009  40 '('           (
   7 8.00010  41 ')'           )
   7 8.00012 123 '{'           {
   7 9.00004 262 TOK_INT       int
   7 9.00008 282 TOK_IDENT     prime
   7 9.00014  61 '='           =
   7 9.00016 283 TOK_INTCON    2
   7 9.00017  59 ';'           ;
   7 10.00004 262 TOK_INT       int
   7 10.00008 282 TOK_IDENT     index
   7 10.00014  61 '='           =
   7 10.00016 283 TOK_INTCON    2
   7 10.00017  59 ';'           ;
   7 11.00004 271 TOK_ARRAY     array
   7 11.00009  60 '<'           <
   7 11.00010 262 TOK_INT       int
   7 11.00013  62 '>'           >
   7 11.00015 282 TOK_IDENT     sieve
   7 11.00021  61 '='           =
   7 11.00023 273 TOK_ALLOC     alloc
   7 11.00028  60 '<'           <
   7 11.00029 271 TOK_ARRAY     array
   7 11.00034  60 '<'           <
   7 11.00035 262 TOK_INT       int
   7 11.00038  62 '>'           >
   7 11.00039  62 '>'           >
   7 11.00040  40 '('           (
   7 11.00041 283 TOK_INTCON    100
   7 11.00044  41 ')'           )
   7 11.00045  59 ';'           ;
   7 13.00004 267 TOK_WHILE     while
   7 13.00010  40 '('           (
   7 13.00011 282 TOK_IDENT     index
   7 13.00017  60 '<'           <
   7 13.00019 283 TOK_INTCON    100
   7 13.00022  41 ')'           )
   7 13.00024 123 '{'           {
   7 14.00007 282 TOK_IDENT     sieve
   7 14.00012  91 '['           [
   7 14.00013 282 TOK_IDENT     index
   7 14.00018  93 ']'           ]
   7 14.00020  61 '='           =
   7 14.00022 283 TOK_INTCON    1
   7 14.00023  59 ';'           ;
   7 15.00007 282 TOK_IDENT     index
   7 15.00013  61 '='           =
   7 15.00015 282 TOK_IDENT     index
   7 15.00021  43 '+'           +
   7 15.00023 283 TOK_INTCON    1
   7 15.00024  59 ';'           ;
   7 16.00004 125 '}'           }
   7 18.00004 267 TOK_WHILE     while
   7 18.00010  40 '('           (
   7 18.00011 282 TOK_IDENT     prime
   7 18.00017  60 '<'           <
   7 18.00019 283 TOK_INTCON    100
   7 18.00022  41 ')'           )
   7 18.00024 123 '{'           {
   7 19.00007 265 TOK_IF        if
   7 19.00010  40 '('           (
   7 19.00011 282 TOK_IDENT     sieve
   7 19.00016  91 '['           [
   7 19.00017 282 TOK_IDENT     prime
   7 19.00022  93 ']'           ]
   7 19.00023  41 ')'           )
   7 19.00025 123 '{'           {
   7 20.00010 282 TOK_IDENT     index
   7 20.00016  61 '='           =
   7 20.00018 282 TOK_IDENT     prime
   7 20.00024  42 '*'           *
   7 20.00026 283 TOK_INTCON    2
   7 20.00027  59 ';'           ;
   7 21.00010 267 TOK_WHILE     while
   7 21.00016  40 '('           (
   7 21.00017 282 TOK_IDENT     index
   7 21.00023  60 '<'           <
   7 21.00025 283 TOK_INTCON    100
   7 21.00028  41 ')'           )
   7 21.00030 123 '{'           {
   7 22.00013 282 TOK_IDENT     sieve
   7 22.00018  91 '['           [
   7 22.00019 282 TOK_IDENT     index
   7 22.00024  93 ']'           ]
   7 22.00026  61 '='           =
   7 22.00028 283 TOK_INTCON    0
   7 22.00029  59 ';'           ;
   7 23.00013 282 TOK_IDENT     index
   7 23.00019  61 '='           =
   7 23.00021 282 TOK_IDENT     index
   7 23.00027  43 '+'           +
   7 23.00029 282 TOK_IDENT     prime
   7 23.00034  59 ';'           ;
   7 24.00010 125 '}'           }
   7 25.00007 125 '}'           }
   7 26.00007 282 TOK_IDENT     prime
   7 26.00013  61 '='           =
   7 26.00015 282 TOK_IDENT     prime
   7 26.00021  43 '+'           +
   7 26.00023 283 TOK_INTCON    1
   7 26.00024  59 ';'           ;
   7 27.00004 125 '}'           }
   7 29.00004 282 TOK_IDENT     index
   7 29.00010  61 '='           =
   7 29.00012 283 TOK_INTCON    2
   7 29.00013  59 ';'           ;
   7 30.00004 267 TOK_WHILE     while
   7 30.00010  40 '('           (
   7 30.00011 282 TOK_IDENT     index
   7 30.00017  60 '<'           <
   7 30.00019 283 TOK_INTCON    100
   7 30.00022  41 ')'           )
   7 30.00024 123 '{'           {
   7 31.00007 265 TOK_IF        if
   7 31.00010  40 '('           (
   7 31.00011 282 TOK_IDENT     sieve
   7 31.00016  91 '['           [
   7 31.00017 282 TOK_IDENT     index
   7 31.00022  93 ']'           ]
   7 31.00023  41 ')'           )
   7 31.00025 123 '{'           {
   7 32.00010 282 TOK_IDENT     putint
   7 32.00017  40 '('           (
   7 32.00018 282 TOK_IDENT     index
   7 32.00023  41 ')'           )
   7 32.00024  59 ';'           ;
   7 33.00010 282 TOK_IDENT     putchr
   7 33.00017  40 '('           (
   7 33.00018 284 TOK_CHARCON   '\n'
   7 33.00022  41 ')'           )
   7 33.00023  59 ';'           ;
   7 34.00007 125 '}'           }
   7 35.00007 282 TOK_IDENT     index
   7 35.00013  61 '='           =
   7 35.00015 282 TOK_IDENT     index
   7 35.00021  43 '+'           +
   7 35.00023 283 TOK_INTCON    1
   7 35.00024  59 ';'           ;
   7 36.00004 125 '}'           }
   7 38.00004 268 TOK_RETURN    return
   7 38.00011 283 TOK_INTCON    0
   7 38.00012  59 ';'           ;
   7 39.00001 125 '}'           }
