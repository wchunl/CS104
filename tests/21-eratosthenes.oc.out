; 0.  1: # 1 "21-eratosthenes.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "21-eratosthenes.oc"
; 4.  1: 
; 4.  2: 
; 4.  3: # 1 "oclib.h" 1
; 5.  1: # 22 "oclib.h"
; 6. 22: void fail (string expr, string file, int line);
; 6. 23: 
; 6. 24: void putchr (int chr);
; 6. 25: void putint (int num);
; 6. 26: void putstr (string str);
; 6. 27: 
; 6. 28: int getchr();
; 6. 29: string getstr();
; 6. 30: string getln();
; 6. 31: # 4 "21-eratosthenes.oc" 2
; 7.  4: 
; 7.  5: 
; 7.  6: 
; 7.  7: 
; 7.  8: int main() {
; 7.  9:    int prime = 2;
; 7. 10:    int index = 2;
; 7. 11:    array<int> sieve = alloc<array<int>>(100);
; 7. 12: 
; 7. 13:    while (index < 100) {
; 7. 14:       sieve[index] = 1;
; 7. 15:       index = index + 1;
; 7. 16:    }
; 7. 17: 
; 7. 18:    while (prime < 100) {
; 7. 19:       if (sieve[prime]) {
; 7. 20:          index = prime * 2;
; 7. 21:          while (index < 100) {
; 7. 22:             sieve[index] = 0;
; 7. 23:             index = index + prime;
; 7. 24:          }
; 7. 25:       }
; 7. 26:       prime = prime + 1;
; 7. 27:    }
; 7. 28: 
; 7. 29:    index = 2;
; 7. 30:    while (index < 100) {
; 7. 31:       if (sieve[index]) {
; 7. 32:          putint (index);
; 7. 33:          putchr ('\n');
; 7. 34:       }
; 7. 35:       index = index + 1;
; 7. 36:    }
; 7. 37: 
; 7. 38:    return 0;
; 7. 39: }
hello